debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbSnapshotChangeEventSource.java

```diff
@@ -157,34 +147,30 @@ protected SnapshotResult<MongoDbOffsetContext> doExecute(ChangeEventSourceContex
     }
 
     @Override
-    protected SnapshottingTask getSnapshottingTask(MongoDbPartition partition, MongoDbOffsetContext previousOffset) {
-        if (previousOffset == null) {
-            LOGGER.info("No previous offset has been found");
-            if (connectorConfig.getSnapshotMode().equals(MongoDbConnectorConfig.SnapshotMode.NEVER)) {
-                LOGGER.info("According to the connector configuration, no snapshot will occur.");
-                return new MongoDbSnapshottingTask(Collections.emptyList());
-            }
-            return new MongoDbSnapshottingTask(replicaSets.all());
-        }
-
-        // Even if there are previous offsets, if no snapshot should occur, return task with no replica sets
+    protected SnapshottingTask getSnapshottingTask(MongoDbPartition partition, MongoDbOffsetContext offsetContext) {
+        // If no snapshot should occur, return task with no replica sets
         if (connectorConfig.getSnapshotMode().equals(MongoDbConnectorConfig.SnapshotMode.NEVER)) {
             LOGGER.info("According to the connector configuration, no snapshot will occur.");
             return new MongoDbSnapshottingTask(Collections.emptyList());
         }
 
+        if (offsetContext == null) {
+            LOGGER.info("No previous offset has been found");
+            return new MongoDbSnapshottingTask(replicaSets.all());
+        }
+
         // Collect which replica-sets require being snapshotted
         final List<ReplicaSet> replicaSetSnapshots = new ArrayList<>();
 
         for (var replicaSet : replicaSets.all()) {
             try (MongoDbConnection mongo = establishConnection(partition, replicaSet, ReadPreference.primaryPreferred())) {
-                final ReplicaSetOffsetContext rsOffsetContext = previousOffset.getReplicaSetOffsetContext(replicaSet);
+                final ReplicaSetOffsetContext rsOffsetContext = offsetContext.getReplicaSetOffsetContext(replicaSet);
                 if (mongo != null && isSnapshotExpected(mongo, rsOffsetContext)) {
                     replicaSetSnapshots.add(replicaSet);
                 }
             }
             catch (InterruptedException e) {
-                throw new DebeziumException("Interrupted while creating snapshotting task", e);
+                throw new RuntimeException(e);
```

Comment:
Why `RuntimeException`?

Reply:
Good catch, I wonder how this got in there...  Likely incorrect rebase